ansible/configs/tech-cockpit-901/pre_software.yml

```diff
@@ -0,0 +1,64 @@
+---
+
+- name: Step 003 Pre Software
+  hosts: localhost
+  connection: local
+  gather_facts: false
+  become: false
+  tasks:
+    - debug:
+        msg: "Step 003 Pre Software"
+
+
+# Cloudformation template or equivalent should tag all hosts with Project:{{ env_type }}-{{ guid }}
+- name: Configure all hosts with Repositories, Common Files and Set environment key
+  hosts: all
+  become: true
+  gather_facts: false
+  tags:
+    - step004
+    - common_tasks
+  roles:
+    - role: set-repositories
+      when: repo_method is defined
+
+    - role: set_env_authorized_key
+      when: set_env_authorized_key | bool
+
+
+- name: Configuring Bastion Hosts
+  hosts: bastions
+  become: true
+  tags:
+    - step004
+    - bastion_tasks
+
+  pre_tasks:
+    - name: Generate strong password
+      set_fact:
+        student_password: "{{ lookup('password', '/dev/null length=12 chars=ascii_letters,digits') }}"
+
+  roles:
+    - role: common
+      when: install_common | default(true) | bool
+
+    - role: bastion
+      when: install_bastion | default(true) | bool
+
+    - role: bastion-student-user
+      when: install_student_user | default(true) | bool
+
+    - role: control-user
```

Comment:
no use of control-user role so should be

Reply:
removed